src/main/java/com/uber/cadence/internal/worker/PollDecisionTaskDispatcher.java

```diff
@@ -0,0 +1,52 @@
+/*
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.uber.cadence.internal.worker;
+
+import com.uber.cadence.PollForDecisionTaskResponse;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class PollDecisionTaskDispatcher implements Consumer<PollForDecisionTaskResponse> {
```

Comment:
This will be instantiated at the factory level and workers will subscribe to it when sticky execution is enabled. internal would be a useful concept or i could move the class up to the worker package